### PR TITLE
FindRefs 'preview' versus 'navigate to source location'.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes: https://devdiv.visualstudio.com/DevDiv/_workitems?id=402396&fullScreen=true&_a=edit

This changes makes it so our FindRefs reuslts follow the expected interaction patterns in VS for a results window.  Specifically:

1. Arrowing through results should show the item in the provisional tab, and place the caret at the location of the result, but not give focus to the text-editor portion.  That way users can continue using the arrow keys to see other results in the FAR window instead of having the arrow keys now navigate in their code.
2. Double clicking should open the item into a non-provisional tab, place the caret at the right location, and also give focus to the editor.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by sharwell** [comment]
@CyrusNajmabadi  Can you clarify the behavior before this change for users that don't have access to view that link?

**Comment by CyrusNajmabadi** [comment]
Arrowing between items would place focus into the editor, thus making it so you could not effectively use arrow keys to move between results.

File: src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs [hunk 1]
```diff
@@ -275,7 +275,11 @@ private static Document OpenDocument(Workspace workspace, DocumentId documentId,
             {
                 if (options.GetOption(NavigationOptions.PreferProvisionalTab))
                 {
-                    using (NewDocumentStateScope ndss = new NewDocumentStateScope(__VSNEWDOCUMENTSTATE.NDS_Provisional, VSConstants.NewDocumentStateReason.Navigation))
+                    // If we're just opening the provisional tab, then do not "activate" the document
+                    // (i.e. don't give it focus).  This way if a user is just arrowing through a set 
+                    // of FindAllReferences results, they don't have their cursor placed into the document.
+                    var state = __VSNEWDOCUMENTSTATE.NDS_Provisional | __VSNEWDOCUMENTSTATE.NDS_NoActivate;
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs]: :question: What happens in the following scenario:

1. Open a document
2. Use the Find All References feature on a local variable with two or more references
3. Close the first document and open a different document
4. Click on a reference in the Find All References window. A provisional tab should appear.
5. Click on the second document which is already open, thus deactivating but not closing the provisional tab.
6. Click the second reference. The provisional tab should be activated again but shouldn't have focus. ⬅️ I'm concerned this step might not be working.

Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs]: Have tested this scenario.  It works exactly as you want it to.

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
tagging @MattGertz 

**Customer scenario**

user uses the new streaming-find-refs window, and then uses arrow keys to try to choose between the items.  When the user hits up/down the preview window opens to show them a preview of the code.  However, the focus also switches to this window, capturing the caret, and making it so that up/down no longer move between FAR results, and instead moves teh caret in the window.

This is annoying, and is a regression over the behavior of the previous FAR experience we used to ship.

**Bugs this fixes:**

https://devdiv.visualstudio.com/DevDiv/_workitems?id=402396&fullScreen=true&_a=edit

**Workarounds, if any**

None.

**Risk**

Low.

**Performance impact**

None.

**Is this a regression from a previous update?**

Yes.  

**Root cause analysis:**

Internally it seems like everyone on the team either uses a mouse to choose items, or uses F8 to cycle between entries.  Boht of these scenarios continue to work.  It was only arrow-keys that changed.

**How was the bug found?**

Customer reported.

**Pull request merged**